node/node.go

```diff
@@ -775,8 +791,10 @@ func NewNode(config *cfg.Config,
 	// Create the handshaker, which calls RequestInfo, sets the AppVersion on the state,
 	// and replays any blocks as necessary to sync CometBFT with the app.
 	consensusLogger := logger.With("module", "consensus")
-	if !stateSync {
-		if err := doHandshake(stateStore, state, blockStore, genDoc, eventBus, proxyApp, consensusLogger); err != nil {
+
+	// RPC recovery is required if blockStore Height is 0
+	if blockStore.Height() == 0 || !stateSync {
```

Comment:
This is worth discussing since there are gaps in knowledge that need to be filled.  There are three cases where the comet height is zero.

1) Actually genesis
2) Statesync completely over the network as we use today
3) LocalStatesync where the application.db is already restored, but comet is at 0

One of the cosmos-sdk to comet integration challenges for local state sync is communicating the height of appstate/application.db

In the initial implementation, an extra applicationd.db height parameter was passed from cosmos-sdk to comet.  However a previous code review tried to eliminate that extra parameter and simply check if comet height was 0, and start RPC statesync.

However that actually causes some issues because a normal network statesync also has comet height == 0, but application.db is also 0. 

We need an agreeable way to determine the height of application.db from cosmos-sdk.  Since a previous code review determined a parameter was not desirable, is there a better way to query the application height?  Is there an object we should look at, or make an ABCI call?

Reply:
I see, thanks for the detailed notes and history here, I wasn't aware of the subtlety here!

> Is there an object we should look at, or make an ABCI call?

An ABCI call (`info`) would indeed provide the app height I believe. But not sure that's the best way.

Reply:
Originally it was added as a variable to configure when cosmos-sdk created the Node{}, but it was asked to be removed during code-review, probably to avoid creating a new dependency between cosmos-sdk and comet

Now that we know the removal causes an issue, it's up to us to pick a new way to compare the heights of comet vs cosmos-sdk app. 

Is there general guidance on how to communicate variables between comet and cosmos-sdk?